maestro-android/src/main/java/dev/mobile/maestro/handlers/LocaleSettingHandler.kt

```diff
@@ -0,0 +1,68 @@
+package dev.mobile.maestro.handlers
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.res.Configuration
+import android.os.Build
+import android.util.Log
+import java.lang.reflect.InvocationTargetException
+import java.util.*
+
+class LocaleSettingHandler(context: Context) : AbstractSettingHandler(context, listOf(CHANGE_CONFIGURATION)) {
+    fun setLocale(locale: Locale) {
+        try {
+            if (hasPermissions()) {
+                setLocaleWith(locale)
+            }
+        } catch (e: Exception) {
+            Log.e(TAG, "Failed to set locale", e)
+        }
+    }
+
+    @SuppressLint("PrivateApi")
+    @Throws(
+        ClassNotFoundException::class,
+        NoSuchMethodException::class,
+        InvocationTargetException::class,
+        IllegalAccessException::class,
+        NoSuchFieldException::class
+    )
+    private fun setLocaleWith(locale: Locale) {
+        var activityManagerNativeClass = Class.forName("android.app.ActivityManagerNative")
```

Comment:
What happens if something here errors out, maybe reflection failed. How would that behaviour look like in terms of error propagation?